spec/integration/aws_subnet_spec.rb

```diff
@@ -41,11 +43,11 @@
             network_acl 'test_network_acl'
           end
         }.to create_an_aws_subnet('test_subnet',
```

Comment:
Had an idea! I think we can just use multiple matchers for this:

```
}.to create_an_aws_subnet('test_subnet',
  :vpc_id => test_vpc.aws_object.id,
  :cidr_block => '10.0.0.0/24',
  :availability_zone => az
).and have_an_aws_subnet('test_subnet',
  :subnet_id => driver.ec2_client.describe_route_tables(:filters=>[{:name=>"route-table-id",:values=>[rt]}]).route_tables[0].associations[0].subnet_id
).and have_an_aws_subnet('test_subnet',
  :subnet_id => driver.ec2_client.describe_network_acls(:filters=>[{:name=>"network-acl-id",:values=>[na]}]).network_acls[0].associations[0].subnet_id
).and be_idempotent
```

Here we are leverging the `create_an*` matcher and the `have_an*` matcher. The `create_an*` matchers verify that the resource creates an aws_object _and_ validates the object properties. The `have_an*` matchers just validate the object properties (the has provided to the method).